torch/distributed/_composable/fsdp/_fsdp_state.py

```diff
@@ -317,12 +314,9 @@ def _register_pre_backward_hook(self, output: Any) -> Any:
         if not torch.is_grad_enabled():
             return output
         flat_outputs, _ = tree_flatten(output)
-        tensors = tuple(
-            t for t in flat_outputs if (torch.is_tensor(t) and t.requires_grad)
-        )
-        if tensors:
-            for tensor in tensors:
-                tensor.register_hook(self._pre_backward)
+        for t in flat_outputs:
+            if torch.is_tensor(t) and t.requires_grad:
+                t.register_hook(self._pre_backward)
```

Comment:
This change is unrelated to wait_event, but it removes usage of `tuple()` generator which is not supported well by Dynamo (maybe with added benefit of slightly simplifying the code).